lib/puppet/provider/package/yum.rb

```diff
@@ -86,23 +86,24 @@ def self.check_updates(disablerepo, enablerepo, disableexcludes)
   end
 
   def self.parse_updates(str)
-    # Strip off all content before the first blank line
-    body = str.partition(/^\s*\n/m).last
+    # Strip off all content that contains Obsoleting, Security: or Update
+    body = str.partition(/^(Obsoleting|Security:|Update)/).first
 
     updates = Hash.new { |h, k| h[k] = [] }
-    body.split.each_slice(3) do |tuple|
-      break if tuple[0] =~ /^(Obsoleting|Security:|Update)/
-      break unless tuple[1] =~ /^(?:(\d+):)?(\S+)-(\S+)$/
-      hash = update_to_hash(*tuple[0..1])
-      # Create entries for both the package name without a version and a
-      # version since yum considers those as mostly interchangeable.
-      short_name = hash[:name]
-      long_name  = "#{hash[:name]}.#{hash[:arch]}"
-
-      updates[short_name] << hash
-      updates[long_name] << hash
-    end
 
+    body.split(/^\s*\n/).each do |line|
+      line.split.each_slice(3) do |tuple|
+        next unless tuple[0].include?('.') && tuple[1] =~ /^(?:(\d+):)?(\S+)-(\S+)$/
```

Comment:
As the same regexp is used https://github.com/puppetlabs/puppet/blob/694a2a8e1df062d303caebb6f29f95a8bfa0ccc2/lib/puppet/provider/package/yum.rb#L121 you could extract it to a constant